updater

```diff
@@ -433,49 +444,45 @@ Refreshes the Pi-Apps details for apps that do not need to be reinstalled for th
   
 }
 
-update_app() { #first arg is app name
+update_app() { #given app name, uninstall it, update app folder, install it again
   local app="$1"
   [ -z "$app" ] && error "update_app(): no app specified!"  
   status "Updating \e[1m${app}\e[0m\e[96m..."
   echo
   
   #Set terminal title
   echo -ne "\e]0;Updating ${app}\a"
-
+  
   # check if update app folder exists before doing anything
   # it can happen that executing the updater from the pi-apps GUI the update folder is missing
   # if the user has no internet or internet issues the update folder will be removed due to the failed git pull and then wait in the loop for a new git clone
   # the GUI updater executes check_repo with the "fast" option so it skips updating the pi-apps update folder and obtains any previously determined updatable apps and files
   # if we do not check for this then apps will be removed, the older version moved to trash, and then the new version will fail to copy over since it does not exist
   if [ -d "${DIRECTORY}/update/pi-apps/apps/${app}" ]; then
-    local installback=no
-    if will_reinstall "$app";then
-      installback=yes
-      status "$app's install script has been updated. Reinstalling $app..."
-      #uninstall it
+    
+    status "$app's install script has been updated. Reinstalling $app..."
+    #uninstall it - if retrying an update with successful uninstall and failed install, don't uninstall again
+    if [ "$(app_status "$app")" != uninstalled ];then
       "${DIRECTORY}/manage" uninstall "$app" update #report to the app uninstall script that this is an uninstall for the purpose of updating by passing "update"
-      
-      #fix edge case: if app is installed but uninstall script doesn't exist somehow, then pretend app was uninstalled so that the reinstall later will happen noninteractively
-      if [ "$(app_status "$app")" == installed ];then
-        echo 'uninstalled' > "${DIRECTORY}/data/status/${app}"
-      fi
     fi
-
+    
+    #fix edge case: force-pretend app was uninstalled so that the reinstall later will happen noninteractively
+    echo 'uninstalled' > "${DIRECTORY}/data/status/${app}"
```

Comment:
manage installs are allowed to occur as long as the app status is not "disabled". That is the only check we have implemented so setting the status should not matter at all. I think the previous "edge case" was impossible because of this so we should remove this entirely. I have checked this to be true, all status except "Disabled" are allowed to proceed when executing manage install.

Reply:
You seem to be right. I think this was from the distant past where manage install ran everything through a gui validate apps function.

Reply:
Removed in https://github.com/Botspot/pi-apps/pull/2708/commits/e7ae18a5cb75c9fc522ed832d00dfe03ae11a49c